proto/src/logcollector.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package org.eln2.proto.logcollector;
+import "types.proto";
+
+// Everything in this file is still in flux. It will be locked down prior to first alpha release.
+service LogService {
+	rpc Log(LogEntry) returns (LogResponse);
+}
+
+message LogEntry {
+	// Fields that identify the client:
+	Version minecraft = 1;
+	Version forge = 2;
+	Version eln = 3;
+	OS os = 4;
+	string uuid = 5;  // May be empty.
+	// The message(s) that we're trying to log.
+	// Any of these fields may be empty.
+	CrashReport crash_report = 6;
+	Analytics analytics = 7;
+}
+
+// This is a straight textual copy of Forge's crash report, for later perusal.
+message CrashReport {
+	string report = 1;
+}
+
+// There should be a config section to disable/limit this.
+message Analytics {
+	// Included mainly for stability/crash analysis.
+	message OtherMod {
+		string name = 1;
+		Version version = 2;
+	}
+	repeated OtherMod mods = 1;
+	// Included for performance analysis.
+	message Circuit {
+		int32 matrix_size = 1;
+		int32 avg_time_usec = 2;
+		repeated Component component = 3;
+	}
+	message Component {
+		string type = 1;
+		int32 count = 2;
+	}
```

Comment:
Would it be faster/more useful to just dump the entire connection matrix, with values for the connected components? That would make it so that we can replicate the state in testing and find things like singular matrices in commonly made circuits.

My guess is that would be undoubtedly more expensive.. but perhaps worth it at least as an opt-in for debugging someone's server if they want (eg, a "My world is having trouble, send current state report using a command"). Not sure.

Reply:
Maybe, but I have no clue how that looks. I'll have to request you provide the code for that one.